hw/misc/hermes.c

```diff
@@ -964,20 +993,37 @@ static void hermes_cleanup_msix(HermesState *hermes)
 
 static void hermes_instance_init(Object *obj)
 {
+    int i;
     HermesState *hermes = HERMES(obj);
     memcpy(&hermes->bar2, &bar2_init, sizeof(bar2_init));
+
+    for (i = 0; i < H2C_CHANNELS; i++) {
+        memcpy(&hermes->bar2.h2c[i], &h2c_init, sizeof(h2c_init));
+        hermes->bar2.h2c[i].irq_vector = i;
+        hermes->bar2.h2c[i].ch = i;
+        hermes->bar2.h2c[i].channel.identifier |= ((i << 8) & 0xF00);
+        sprintf(hermes->bar2.h2c[i].name, "h2c-dma-%u", i);
+    }
+
+    for (i = 0; i < C2H_CHANNELS; i++) {
+        memcpy(&hermes->bar2.c2h[i], &c2h_init, sizeof(c2h_init));
+        hermes->bar2.c2h[i].irq_vector = H2C_CHANNELS + i;
+        hermes->bar2.c2h[i].ch = i;
+        hermes->bar2.c2h[i].channel.identifier |= ((i << 8) & 0xF00);
+        sprintf(hermes->bar2.h2c[i].name, "c2h-dma-%u", i);
```

Comment:
Ops, only noticed this now. This should be c2h, not h2c

Reply:
Ah - good catch!